lightning/src/ln/channelmanager.rs

```diff
@@ -16640,6 +16644,20 @@ where
 							},
 						}
 					}
+					for (htlc_source, htlc) in monitor.get_onchain_failed_outbound_htlcs() {
+						log_info!(
+							args.logger,
+							"Failing HTLC with payment hash {} as it was resolved on-chain.",
+							htlc.payment_hash
+						);
+						failed_htlcs.push((
```

Comment:
Will we generate a `PaymentFailed` and/or `HTLCHandlingFailed` event for these HTLCs on every restart until the monitor is removed? Could be worth noting in a comment if so